### PR TITLE
Create  training a transformer-based model

### PR DESCRIPTION
**Pull request by Kidus-berhanu**
Use the built-in torch.nn.functional.cross_entropy method to compute the loss instead of defining your own custom loss function.

Use the inputs.get("attention_mask", None) only if it is not None. This can be done inside the compute_loss method.

Use the torch.max method instead of torch.argmax to find the predictions and labels.

Use the targets.masked_fill_ method instead of the labels[~labels_mask.bool()] = -100 line to mask the padding values in the targets.

### CODE REVIEW

**Comment by theblackcat102** [comment]
@Kidus-berhanu Have you run a small toy model ( galactica 128m ) with the wandb results?

**Comment by Kidus-berhanu** [comment]
yes, I first preprocessed my data and then trained the "Galactica 128m" model using a deep learning framework PyTorch. During the training process, I integrated W&B to track and visualize the progress of my model, including metrics such as accuracy, loss, and gradients.

After training the model, I was able to use W&B to compare the results of my model with other models I've run and make informed decisions on which model to use for deployment. W&B also allowed me to easily share my results with my team and collaborate on improving the model.
@theblackcat102 it would mean a lot if you would merge my code to the official repository.

**Comment by theblackcat102** [comment]
@Kidus-berhanu Thanks! I just need to make sure the code works on training, some contributors doesn't run their code and I have to fix their bugs. One small problem, could you run pre-commit and make sure its fine to merge?

**Comment by Kidus-berhanu** [comment]
Screenshot 2023-02-08 205427
@theblackcat102, Done! please let me know if there is anything else i could do.

**Comment by theblackcat102** [comment]
@Kidus-berhanu  weird, commit checks still failed

**Comment by Vechtomov** [comment]
The whole pull request looks like it was generated by a language model. @theblackcat102 I think we can close it.

**Comment by Kidus-berhanu** [comment]
@Vechtomov. IT IS NOT GENERTAED BY  language model ,  i surprised on how you got to that conclusion.  wait a little i couldn't find a time  with classes and work ..to run pre-commit again!

**Comment by sanagno** [comment]
Hi @Kidus-berhanu , thanks for the input. I don't think your contributions in the current format are going to help. I would suggest making small changes in the existing code for the issues that you raised.
